src/App.tsx

```diff
@@ -14,11 +15,99 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[],
+  isGoodsVisible: boolean,
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: [...goodsFromServer],
+    isGoodsVisible: false,
+  };
+
+  changeButtonState = () => {
+    this.setState((prevState) => ({
+      isGoodsVisible: !prevState.isGoodsVisible,
```

Comment:
you hide this button after clicking, then this button will be never shown, so it no sense to toggle value here, you can just set it to true